src/lib.rs

```diff
@@ -611,37 +578,37 @@ impl Client {
     pub fn build_report(&self) -> ReportBuilder {
         ReportBuilder {
             client: self,
-            send_strategy: None,
         }
     }
 
-    /// Function used internally to send payloads to Rollbar as default `send_strategy`.
-    fn send(&self, payload: String) -> thread::JoinHandle<Option<ResponseStatus>> {
+    /// Function used internally to send payloads to Rollbar.
+    async fn send(&self, payload: String) -> Option<ResponseStatus> {
+        let payload_clone = payload.clone();
         let body = hyper::Body::from(payload);
         let url = std::env::var("ROLLBAR_ENDPOINT").unwrap_or_else(|_| "".to_string());
+        if url.len() == 0 {
+            println!("no rollbar endpoint, not sending payload {}", payload_clone);
+            return None;
+        }
         let request = Request::builder()
             .method(Method::POST)
             .uri(url)
             .body(body)
             .expect("Cannot build post request!");
 
-        let job = self
+        let response_future = self
             .http_client
-            .request(request)
-            .map(|res| Some(ResponseStatus::from(res.status())))
-            .map_err(|error| {
-                println!("Error while sending a report to Rollbar.");
-                print!("The error returned by Rollbar was: {:?}.\n\n", error);
+            .request(request);
 
+        let response_result = response_future.await;
+        match response_result {
+            Ok(response) => Some(ResponseStatus::from(response.status())),
+            Err(e) => {
+                println!("Error while sending a report to Rollbar.");
+                print!("The error returned by Rollbar was: {:?}.\n\n", e);
                 None::<ResponseStatus>
-            });
-
-        thread::spawn(move || {
-            current_thread::Runtime::new()
-                .unwrap()
-                .block_on(job)
-                .unwrap()
-        })
```

Comment:
converted thread spawn move join handle whatever to async await. i have no idea if that is similar or effective.